From 018e5a3739170c15f1f29475940e0be4defdd5f0 Mon Sep 17 00:00:00 2001
From: Lou Viannay <lou.viannay@octavesolution.com>
Date: Tue, 31 Mar 2026 07:49:04 +0800
Subject: [PATCH 62/69] dts: add spidev node on SPI1 for userspace SPI access

Add a generic spidev device on SPI1 CS0 (pins PH5-PH8 on the 40-pin
header). This exposes /dev/spidev1.0 via the spidev driver, allowing
userspace programs to drive arbitrary SPI devices without a custom
kernel driver.

Compatible string rohm,dh2228fv is the standard mainline placeholder
for spidev nodes. Maximum clock rate set to 10 MHz; the spidev driver
allows per-transfer overrides down to any lower rate.

Signed-off-by: Blue Cuenca <blue.cuenca@gmail.com>
---
 arch/arm64/boot/dts/allwinner/sun50i-h618-longanpi-3h.dts | 7 +++++++
 1 file changed, 7 insertions(+)

diff --git a/arch/arm64/boot/dts/allwinner/sun50i-h618-longanpi-3h.dts b/arch/arm64/boot/dts/allwinner/sun50i-h618-longanpi-3h.dts
index 2291f5c46..59cb86c92 100644
--- a/arch/arm64/boot/dts/allwinner/sun50i-h618-longanpi-3h.dts
+++ b/arch/arm64/boot/dts/allwinner/sun50i-h618-longanpi-3h.dts
@@ -227,7 +227,14 @@ &uart1 {
 
 &spi1 {
 	pinctrl-0 = <&spi1_pins>, <&spi1_cs0_pin>;
+	pinctrl-names = "default";
 	status = "okay";
+
+	spidev@0 {
+		compatible = "rohm,dh2228fv";
+		reg = <0>;
+		spi-max-frequency = <10000000>;
+	};
 };
 
 &i2c3 {
-- 
2.50.1

